src/shared_utilities/drivers.jl

```diff
@@ -185,11 +186,117 @@ PrescribedPrecipitation{FT}(liquid_precip) where {FT} =
 
 """
-    CoupledRadiativeFluxes{FT} <: AbstractRadiativeDrivers{FT}
+    CoupledRadiativeFluxes{FT, F <: Union{Function, Nothing}} <: AbstractRadiativeDrivers{FT}
 
-To be used when coupling to an atmosphere model.
+To be used when coupling to an atmosphere model. If F is a function, cosθs is updated in
+the driver update.
 """
-struct CoupledRadiativeFluxes{FT} <: AbstractRadiativeDrivers{FT} end
+struct CoupledRadiativeFluxes{FT, F <: Union{Function, Nothing}} <:
+       AbstractRadiativeDrivers{FT}
+    """Function that returns sun zenith angle in radians given the arguments:
+    (time from simulation start date in seconds, longitude::FT, latitude::FT)"""
+    θs::F
+end
+
+CoupledRadiativeFluxes{FT}() where {FT} =
+    CoupledRadiativeFluxes{FT, Nothing}(nothing)
+
+CoupledRadiativeFluxes(::Type{FT}, args...) where {FT} =
+    CoupledRadiativeFluxes{FT}(args...)
+
+"""
+    CoupledRadiativeFluxes{FT}(
+        start_date::Dates.DateTime,
+        insol_params::Insolation.Parameters.InsolationParameters{FT} = LP.LandParameters(FT,
+        ).insol_params,
+    )
+Creates a `CoupledRadiativeFluxes` object with a default zenith angle function that uses Insolation.jl
+to compute the zenith angle at a given time and location.
+"""
+function CoupledRadiativeFluxes{FT}(
+    start_date::Dates.DateTime,
+    insol_params::Insolation.Parameters.InsolationParameters{FT} = LP.LandParameters(
+        FT,
+    ).insol_params,
+) where {FT}
+    zenith_func = make_zenith_angle(FT; start_date, insol_params)
+    return CoupledRadiativeFluxes{FT, typeof(zenith_func)}(zenith_func)
+end
+
+"""
+    function make_zenith_angle(
+        ::Type{FT};
+        start_date::Dates.DateTime,
+        insol_params::Insolation.Parameters.InsolationParameters{FT},
+        longitude::L = nothing,
+        latitude::L = nothing,
+    )
+
+Returns a zenith angle function for the given start date, insolation parameters,
+and optionally location.
+
+If `longitude` and `latitude` are not supplied, the returned function takes a time in seconds,
+and latitude and longitude in `FT`, and returns the zenith angle at that time and location.
+If `longitude` and `latitude` are supplied, the returned function takes a time in seconds
+and returns the zenith angle at that time, using the supplied longitude and latitude.
+"""
+function make_zenith_angle(
+    ::Type{FT};
+    start_date::Dates.DateTime,
+    insol_params::Insolation.Parameters.InsolationParameters{FT},
+    longitude::L = nothing,
+    latitude::L = nothing,
+) where {FT, L <: Union{Nothing, FT}}
+    if L <: Nothing
+        return let start_date = start_date, insol_params = insol_params
+            function zenith_angle(t::T, long::FT, lat::FT) where {T}
+                if T <: ITime
+                    current_datetime = date(t)
+                else
+                    current_datetime = start_date + Dates.Second(round(t))
+                end
```

Comment:
I think this will work
```suggestion
                current_datetime = T isa ITime ? date(t) : start_date + Dates.Second(round(t))
```